### ISSUE TITLE
Not an issue!

### ISSUE THREAD

**Issue by Guyverix** [CodeS50/slim-framework-4-jwt-auth-example#2]
Just wanted to say thanks for posting this repo!  I have been fighting for 2 weeks to do exactly what you provided.  The examples that I was able to find online, as well as looking at other repos that had a working JWT token were either configured completely differently or were too complex for me to tease out what they were trying to do.

Your example was exactly what I needed.  Once again, thank you for posting this repo.

**Comment by CodeS50** [comment]
I'm very happy to help you. When I shared this repo there was no JWT auth sample for Slim Framework. I also shared it to help the Slim Framework community.

**Comment by Guyverix** [comment]
the fact that you made fully functional examples really helped me a lot!  

I have already started adding things like scopes into the token and am getting ready to test them within my classes.  I notice that your examples were for get function().  I did not see anything for classes in your test cases..

From what I can tell to do things like access on a per user variable for classes, it needs to be done inside the class definition?
for example: 
```
  $app->group('/dump' , function (Group $group) {
      $group->get('/{table}', ViewTableEventAction::class);
  })->add(JwtAuthenticationMiddleware::class);
```
If I wanted to follow down into the Class " ViewTableEventAction::class" where {table} = test, and I have a variable in my token called accessList for each user I am guessing I need to do the validation inside the class itself to see of {table} value is in my array of accessList?

It appears that I cannot put "logic" in the group function, so I am guessing I have to go down further to do additional checks and scope validation, unless there a reasonable way of checking at the route.php level outside of the Class definition for ```get('/{table}', ViewTableEventAction::class) ```?

Basically what I am attempting is
```
if ( ! in_array (<table name>, $accessListArray)) { 
    $response->getBody()->write('Additional access required.  Contact admin.');
     return $response->withStatus(418);
}
else {
  $group->get('/{table}', ViewTableEventAction::class);
}
```
It would be nice to have this logic inside the route, but I am guessing that is simply not possible when calling a new Class within the $app->group?

**Comment by CodeS50** [comment]
It is better to use a new MiddleWare for this.


src/Application/Middleware/AccessListMiddleware.php

```
<?php


namespace App\Application\Middleware;


use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface as Request;
use Psr\Http\Server\MiddlewareInterface as Middleware;
use Psr\Http\Server\RequestHandlerInterface as RequestHandler;
use Slim\Psr7\Response;

class AccessListMiddleware implements Middleware
{
    /**
     * @var string
     */
    private $table_name;

    /**
     * AccessListMiddleware constructor.
     * @param string $table_name
     */
    public function __construct(string $table_name)
    {
        $this->table_name = $table_name;
    }

    /**
     * {@inheritdoc}
     */
    public function process(Request $request, RequestHandler $handler): ResponseInterface
    {
        /*
         * You can keep role values in JWT.
         *   In this way, you can verify as follows.
         *
         * Data assigned with withAttribute in JwtAuthenticationMiddleware (return $request->withAttribute("account", $data);)
         * */

        $jwt_public_data = $request->getAttribute("account");

        if (in_array($this->table_name, $jwt_public_data["accessListArray"])) {
            return $handler->handle($request);
        }

        $response = new Response();
        $response->getBody()->write('Additional access required.  Contact admin.');
        return $response->withStatus(418);
    }
}
```

use in app/routes.php 
```
$group->get('/{table}', ViewTableEventAction::class)->add(new \App\Application\Middleware\AccessListMiddleware("<table name>"))->add(JwtAuthenticationMiddleware::class);
```

NOTE:
"accessListArray" should be added to $token["data"] in the "src/Library/JwtGenerator.php" file.
example:
```
$token["data"]["accessListArray"] = [
  "<table name1>",
  "<table name2>",
  "<table name3>",
];
```
In this way, the user role system is obtained.

**Comment by Guyverix** [comment]
Thats great!  Straightforward enough that I can follow it, and powerful enough that it is a complete solution for a full fledged app..

Thank you for answering my question, and showing me a path forward on this!
